File: JavaScript/stack.js


// //////////////////////////////////// Implementing Stack using javascript array ///////////////////////////////////////

class Stack {

    constructor() {
        this.arr = [];
        this.top = -1;
    }

    // methods
    push(val) {
        this.top++;
        this.arr[this.top] = val;
    }

    pop() {
        if (this.empty()) return "stack is empty";
        this.top--;
    }

    peek() {
        if (this.empty()) return "stack is empty";
        return this.arr[this.top];
    }
    empty() {
        if (this.top < 0) return true;
        return false;
    }
}

const test = new Stack()
test.push(2);
test.push(1);
test.pop();

console.log(test.peek());

test.pop();


console.log(test.empty());

console.log(test.pop());

File: C++/stack.cpp
#include <stdio.h>

#include <iostream>
// #include <stack>
using namespace std;

///////////////////////////////////  Createing Stack using array /////////////////////////////

class Stack
{
public:
    int *arr;
    int size;
    int top;
    Stack(int size)
    {
        this->size = size;
        arr = new int[size];
        top = -1;
    }

    // methods

    void push(int element)
    {
        if (size - top > 1)
        {
            top++;
            arr[top] = element;
        }
        else
            cout << "Stack Overflow" << endl;
    }

    void pop()
    {
        if (top >= 0)
        {
            top--;
        }
        else
            cout << "Stack Underflow" << endl;
    }

    int peek()
    {
        if (top >= 0 && top < size)
            return arr[top];
        else
        {
            cout << "Stack is Empty" << endl;
            return -1;
        }
    }

    bool isEmpty()
    {
        if (top == -1)
            return true;
        else
            return false;
    }
};

int main()
{

    Stack st(5);
    st.push(1);
    st.push(2);
    st.push(3);

    while (!st.isEmpty())
    {
        cout << st.peek() << " ";
        st.pop();
    }

    // stack<int> stack;

    // stack.push(1);
    // stack.push(2);
    // stack.push(3);

    // while (!stack.empty())
    // {
    //     cout << stack.top() << " ";
    //     stack.pop();
    // }

    return 1;
}